plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,655 @@
+8. After period 2, the user who responded to the challenge receives Mallory’s `exit bond`, no one exits any UTXOs.
+
+
+### Mallory attempts to exit a spent input:
+
+1. Mallory spends `UTXO1` and `UTXO2` in `TX1`.
+2. Mallory spends `UTXO1` in `TX2`.
+3. `TX1` and `TX2` are in-flight.
+4. Mallory starts an exit referencing `TX1` and places `exit bond`.
+5. Mallory starts an exit referencing `TX2` and places `exit bond`.
+6. In period 1 of the exit for `TX1`, someone challenges the canonicity of `TX1` by revealing `TX2`.
+7. In period 1 of the exit for `TX2`, someone challenges the canonicity of `TX2` by revealing `TX1`.
+8. After period 2 of the exit for `TX1`, the challenger receives `exit bond`, no one exits any UTXOs.
+9. After period 2 of the exit for `TX2`, the challenger receives `exit bond`, no one exits any UTXOs.
+
+
+### Operator tries to steal funds from an included transaction
+
+1. Alice spends `UTXO1` in `TX1` to Bob, creating `UTXO2`.
+2. `TX1` is included in (valid) block `N`.
+3. Operator creates invalid deposit, creating `UTXO3`.
+4. Operator spends `UTXO3` in `TX3`, creating `UTXO4`.
+5. Operator starts an exit referencing `TX3` and places `exit bond`.
+6. Operator piggybacks onto the exit and places `piggyback bond`.
+7. Bob starts a *standard* exit for `UTXO2`.
+8. Operator’s exit will have priority of position of `UTXO3`.
+Bob’s exit will have priority of position of `UTXO2`.
+9. Bob receives the value of `UTXO2` first, Operator receives the value of `UTXO4` second.
+All bonds are refunded.
+
+
+### Operator tries to steal funds from an in-flight transaction.
+
+1. Alice spends `UTXO1` in `TX1` to Bob, creating `UTXO2`.
+2. `TX1` is in-flight.
+3. Operator creates invalid deposit, creating `UTXO3`.
+4. Operator spends `UTXO3` in `TX3`, creating `UTXO4`.
+5. Operator starts an exit referencing `TX3` and places `exit bond`.
+6. Operator piggybacks onto the exit and places `piggyback bond`.
+7. Bob starts an exit referencing `TX1` and places `exit bond`.
+8. Bob piggybacks onto the exit and places `piggyback bond`.
+9. Alice is honest, so she hasn’t spent `UTXO1` in any transaction other than `TX1`.
+10. Operator’s exit will have priority of position of `UTXO3`.
+Bob’s exit will have priority of position of `UTXO1`.
+11. Bob receives the value of `UTXO2` first, Operator receives the value of `UTXO4` second.
+All bonds are refunded.
+
+
+### Attack Vectors and Mitigations
+
+#### Honest Exit Bond Slashing
+
+It’s possible for an honest user to start an exit and have their exit bond slashed.
+This can occur if one of the inputs to a transaction is malicious and signs a second transaction spending the same input.
+
+The following scenario demonstrates this attack:
+1. Mallory spends `UTXO1` in `TX1` to Bob, creating `UTXO2`.
+2. `TX1` is in-flight.
+3. Operator begins withholding blocks while `TX1` is still in-flight.
+4. Bob starts an exit referencing `TX1` and places `exit bond`.
```

Comment:
This is the same situation as ☝️, right?

> I think it is not rational for Bob to do that, he gains nothing just loses exit_bond. Bob should not process the incoming transaction until he sees it included in the chain.

Reply:
Right.  I think Bob should not consider the funds received, and should not attempt an exit, until he sees it confirmed on-chain.  Whatever he's giving Mallory for this money, he should hold onto it until then.